### ISSUE TITLE
Modify existing new/edit view to include partials for new tabs

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#1374]
### Descriptive summary

Add code to implement the new/edit form views for collections.

### Rationale

See model issue for rationale.

### Implementation

**Files:**
* [app/views/hyrax/collections/new.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/new.html.erb)
* [app/views/hyrax/collections/edit.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/edit.html.erb)
* [app/views/hyrax/collections/_form.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/_form.html.erb)
* and maybe others in the [app/views/hyrax/collections](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections) directory

#### Expected user interactions

Modify the UI view for creating and editing collections to stay editing after create.
- [ ] Modify title above tabs to say New <Collection Type> (e.g. New User Collection) and Edit <Collection Type>: <Collection Name> (e.g. Edit User Collection: My Favorites) 
- [ ] When creating a new collection, only show Description tab (current behavior).  After save, forward to edit action for the collection, which will show all tabs (new behavior).
- [ ] Show hide tabs according to collection type settings
  - [ ] if nestable=true, show Relationships tab
  - [ ] if sharing=true, show Sharing tab
  - [ ] if discovery=true, show Discovery tab
.

Additional changes for this page are described in the changing/new partials.  See those issues for details.  

The partials will set up the form with the following tabs...
- [ ] Description  (#) EDIT 
- [ ] Branding  (#) - NEW
- [ ] Relationships (#) - NEW
- [ ] Sharing  (#) - EDIT
- [ ] Discovery - EDIT

NOTE: The mockup also shows a Visibility tab and Workflow tab.  Those are for advanced features that are not part of the first sprint.  They will not be visible to users at this time.

This issue closes when the tasks listed in this issue and all the partial issues are complete.

#### SPECIAL NOTE on Admin Sets:

For the initial implementaiton, the Admin Set underlying code will not change.  But to a user, the Admin Set will appear to be a collection.

To make the illusion happen...
* new/edit for admin sets will not be served by the views described in this issue.  They will continue to be served by the admin set views [new](https://github.com/samvera/hyrax/tree/master/app/views/hyrax/admin/admin_sets/new.html.erb) and [edit](https://github.com/samvera/hyrax/tree/master/app/views/hyrax/admin/admin_sets/edit.html.erb).  See issues for partials for specific comments on the differences and suggestions for bringing them into closer alignment.

### Mockup

TODO: Mockup is coming soon.

### Related work

| Collections feature | Issue |
| ------------------------ | ----- |
| Routes                   | # |
| Controller               | # |
| Model                    | # |
| View - Index             | # |
| View - Show - All        | # |
| View - Show - Your       | # |
| View - Show - Landing    | # |
| View - New/Edit          | (this issue) |
| Partial -  Description   | # |
| Partial -  Branding      | # |
| Partial -  Relationships | # |
| Partial -  Sharing       | # |
| Partial -  Discovery     | # |

**Comment by elrayle** [comment]
> When creating a new collection, only show Description tab (current behavior). After save, forward to edit action for the collection, which will show all tabs (new behavior).

Expanded description of expected user experience when creating a new collection...
* User clicks New Collection button
* Only Descriptions tab is shown
* User clicks Save
* collection is created and saved
* UI remains in Edit mode with all tabs shown
* User interacts more with new/edit form and clicks Save
* User is forwarded to Collection Show page

**Comment by elrayle** [comment]
NOTE: The collections directory needs to move when the routes are updated in issue #1380.  Will this effect your work in any way beyond the code moving from app/view/hyrax/collections to app/view/hyrax/dashboard/collections?

**Comment by gkostin1966** [comment]
@elrayle I don't think it will affect what I have already done beyond the changing of paths.

**Comment by elrayle** [comment]
Remaining Work:
* New/edit form should show the collection type in the title (e.g. New User Collection, Edit Exhibit: _Collection Title_)

**Comment by elrayle** [comment]
All requirements are met by various PRs related to the sub-issues identified in the issue description.  OR by the issues identified under remaining work.  Closing.
